fvm/src/gas/charge.rs

```diff
@@ -14,6 +18,8 @@ pub struct GasCharge {
     pub compute_gas: Gas,
     /// Storage costs
     pub storage_gas: Gas,
+    /// Execution time related to this charge, if successfully measured.
+    pub elapsed: Arc<OnceCell<Duration>>,
```

Comment:
I'd prefer to hide the details here if possible (and maybe avoid the atomic op on drop?). E.g.:

```rust
pub struct GasDuration(Option<Arc<OnceCell<Duration>>>);

impl GasDuration {
    pub fn get(&self) -> Option<Duration> {
        self.0.as_ref().and_then(|d|d.get())
    }
}
```